codegen/src/api/storage.rs

```diff
@@ -300,21 +315,32 @@ fn generate_storage_entry_fns(
 
     let client_fns = quote! {
         #docs_token
-        pub async fn #fn_name(
+        pub fn #fn_name(
             &self,
             #( #key_args, )*
```

Comment:
Note to self; key args have a lifetime of 'a, which is the same as the lifetime of the client struct.

Should key args have a lifetime of 'b, and the returned future rely on both 'a and 'b, to avoid any unnecessary restrictions on how the future is used in conjunction with these references?

Reply:
I did a little testing and it seems OK to use the `'a` lifetime for key_args and client; from reading around I think Rust will pick the correct smallest lifetime that all params with `'a` use anyway, so no need to add other lifetimes.